### ISSUE TITLE
Add refs (as in !) to documentation

### ISSUE THREAD

**Issue by smimram** [savonet/liquidsoap#23]
From the ML:

While working with liquidsoap i found some confusing (for me) things:
What means "!" symbol before variable? As i understood, it is not negation (found this in dynamic sources list http://savonet.sourceforge.net/doc-svn/dynamic_sources.html).
This somehow changes logic working with reference-variables (defined with ref?)

Is it possible to make "ref source" variable to use it later?

If i have playlist and harbor, like radio = fallback([live, playlist]), will playlist be paused when somebody is connected on harbor? If not, are there any benefits to pause it? Is it possible to pause it?

@dbaelde answered:

You're right to ask, this (and other things) is missing from
  http://savonet.sourceforge.net/doc-svn/language.html

In short, a reference is a cell whose content can be updated. This is
how you store a value that needs to be changed later. You create a
reference with initial value V using ref(V), you access the current
content or a reference R using !R, and you update it using R := V'.

The content of a reference can have any type. However, storing sources
in references can have unintended consequences, and most often it
won't do what you think it would.

For example, if you define the source switch([!R,blah]) where R is a
reference, then the first child of the switch will be the current
content of the reference (the switch doesn't even know the source came
from a reference) so subsequent updates of the reference won't have
any effect on your stream.

**Comment by mcfiredrill** [comment]
I feel like I could write something on this , and as far as I know there still isn't a whole lot of documentation around refs. The question for me is where to put it. Perhaps in here
http://liquidsoap.fm/doc-svn/language.html

As well as the API reference itself.
